architecture/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,467 @@
+
+### Scenario 2: Amy contributes and open-sources PostgreSQL support to Radius
+
+Amy is a system integrator who helps customers build cloud native applications on AWS. Amy heard about Radius and starts integrating their customer workloads in Radius. She sees a lot of her scenarios involve PostgreSQL and finds Radius doesn’t support that yet. She wants to contribute the PostgreSQL support to Radius and open source it so anyone can use it
+
+### Scenario 3: Raj publishes Recipes for proprietary services
+
+Raj is a service provider who works at Oracle Database and helps enterprises integrate oracle database to their applications. Raj hears about Radius and wants to write Recipes to create an Oracle Database with a standard set of best practices and policies integrated. He wants to provide this Recipes to his customers for easy entry and integration with Radius.
+
+To keep the scope of the document simple, we will focus on the first scenario in this document. The other scenarios will be covered in the child documents.
+
+## Key dependencies and risks
+<!-- What dependencies must we take in order to enable this scenario? -->
+
+<!-- What other risks are you aware of that need to be mitigated. If you have a mitigation in mind, summarize here. -->
+<!-- **Dependency Name** – summary of dependency.  Issues/concerns/risks with this dependency -->
+<!-- **Risk Name** – summary of risk.  Mitigation plan if known. If it is not yet known, no problem. --> 
+
+- Risk: Adoption and community engagement
+    - There is a risk that users may not fully embrace the extensibility feature because its too complex or not powerful enough.
+    - Mitigation: Design with the guiding principle of making it simple and easy to use ; Actively engage with users early-on, provide clear documentation, tutorials and resources.
+
+- Risk: Compatibility and maintainability
+    - As more custom resource types are added to Radius, there is a risk of compatibility issues and increased maintenance overhead.
+    - Mitigation: Implement thorough testing and versioning strategies to ensure compatibility, and establish clear guidelines for maintenance and ownership of the custom resource types.
+
+- Risk: Security and trust
+    - Allowing users to contribute custom resource types introduces potential security risks and the need to establish trust in the community-contributed code.
+    - Mitigation: Implement strict security measures with handling secrets, functional tests and establish a transparent review process for community contributions.
```

Comment:
Is this different from the mechanisms we already have in place? This is applicable to any community contribution, so I'm wondering if the callout refers to an increased risk surface area or something else.

Reply:
The mitigation is for the new repo for open-sourcing custom resource types